phalcon/mailer.zep

```diff
@@ -0,0 +1,513 @@
+		let cName = trim(preg_replace("/[\r\n]+/", "", cName));
+		let this->cc[] = "\"" . cName . "\" <" . c . ">";
+	}
+
+	public function clearCC() {
+		unset(this->cc);
+		let this->cc = [];
+	}
+
+	public function addBCC(bc, bcName = "") {
+		let bcName = trim(preg_replace("/[\r\n]+/", "", bcName));
+		let this->bcc[] = "\"" . bcName . "\" <" . bc . ">";
+	}
+
+	public function clearBCC() {
+		unset(this->bcc);
+		let this->bcc = [];
+	}
+	
+	public function addAttachment(filePath) {
+		let this->attachments[] = filePath;
+	}
+
+	public function clearAttachments() {
+		unset(this->attachments);
+		let this->attachments = [];
+	}
+
+	public function isHTML(){
+		let this->contentType = "text/html";
+	}
+
+	private function multipartMessage(htmlpart, boundary) {
+		var parts, altBoundary, i, attachment, filename, ext, message;
+		
+		if(this->altBody == "") { let this->altBody = this->strip_html_tags(htmlpart); }
+		
+		//let altBoundary = this->generateBoundary();
+		let altBoundary = boundary;
+		ob_start(); //Turn on output buffering
+		let parts = "";
+		//let parts  = "This is a multi-part message in MIME format." . this->newline . this->newline;
+		//let parts = "--" . boundary . this->newline;
+		
+		//let parts .= "Content-Type: multipart/alternative;" . this->newline;
+		//let parts .= "    boundary=\"" . altBoundary . "\"" . this->newline . this->newline;
+
+		let parts .= "--" . altBoundary . this->newline;
+		let parts .= "Content-Type: text/plain; charset=". this->charset . this->newline;
+		let parts .= "Content-Transfer-Encoding: ". this->transferEncodeing . this->newline . this->newline;
+		let parts .= this->altBody . this->newline . this->newline;
+
+		let parts .= "--" . altBoundary . this->newline;
+		let parts .= "Content-Type: text/html; charset=" . this->charset . this->newline;
+		let parts .= "Content-Transfer-Encoding: " . this->transferEncodeing . this->newline . this->newline;
+		let parts .= htmlpart . this->newline . this->newline;
+		
+		//let parts .= "--" . altBoundary . "--" . this->newline . this->newline;
+		
+		if count(this->attachments) > 0 {
+			for i in range(0, count(this->attachments) - 1){
+				let attachment = chunk_split(base64_encode(file_get_contents(this->attachments[i])));
+				let filename = basename(this->attachments[i]);
+				let ext = pathinfo(filename, PATHINFO_EXTENSION);
+				let parts .= "--" . boundary . this->newline;
+				let parts .= "Content-Type: application/" . ext . "; name=\"" . filename . "\"" . this->newline;
+				let parts .= "Content-Transfer-Encoding: base64" . this->newline;
+				let parts .= "Content-Disposition: attachment; filename=\"" . filename . "\"" . this->newline . this->newline;
+				let parts .= attachment . this->newline;
+			}
+		}
+		
+		let parts .= "--" . boundary . "--";
+		
+		let message = ob_get_clean(); //Turn off output buffering
+		return parts;
+	}
+	
+	private function generateBoundary() {
+		return md5(uniqid(time()));
+		/*
```

Comment:
remove those comments and others. Also i think we have some service for generating something like this.